finstmt/findata/database.py

```diff
@@ -43,16 +43,34 @@ def _repr_html_(self):
             lambda x: f"${x:,.0f}" if not x == 0 else " - "
         )._repr_html_()
 
+    def __repr__(self) -> str:
+        statement_items: dict = cast(dict, self.statement_items)
+        return json.dumps(
+            {
+                k: v.get_value(self)
+                for (k, v) in statement_items.items()
+                if v.get_value(self) != 0 and v.item_config.show_on_statement
+            },
+            indent=2,
+        )
```

Comment:
Ah you're right, because the statement item needs the statement to resolve the value we can't have that kind of setup. I think this is the best we can do then. Could you just refactor it so that it only calls `get_value` once per statement item? Right now it's calculating it twice, once for the if condition and again for the final value. 

To do that, I think you'll either need to move away from the dict comprehension, or you can [use the walrus operator](https://realpython.com/python-walrus-operator/#list-comprehensions) since we require Python >=3.8.